src/java/org/broadinstitute/dropseqrna/utils/ValidateAlignedSam.java

```diff
@@ -0,0 +1,129 @@
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.broadinstitute.dropseqrna.utils;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.dropseqrna.cmdline.DropSeq;
+
+import htsjdk.samtools.*;
+import htsjdk.samtools.util.IOUtil;
+import htsjdk.samtools.util.Log;
+
+import picard.cmdline.CommandLineProgram;
+import picard.cmdline.StandardOptionDefinitions;
+
+
+@CommandLineProgramProperties(summary = "Validate BAM file(s)",
+        oneLineSummary = "Validate each BAM file in the input to make sure that\n" +
+        "- It exists and is readable\n" +
+        "- Sort order matches the specified one (default: coordinate)\n" +
+        "- There is a non-zero sequences in the sequence dictionary\n" +
+        "- Sequence dictionaries for all input BAM(s) are identical\n" +
+        "- There are non-zero # of reads\n" +
+        "- Optionally, there are non-zero # of paired reads",
+        programGroup = DropSeq.class)
+
+public class ValidateAlignedSam extends CommandLineProgram {
+
+    private static final Log log = Log.getInstance(ValidateAlignedSam.class);
+
+    @Argument(shortName = StandardOptionDefinitions.INPUT_SHORT_NAME, doc = "The input SAM or BAM file(s) to validate. This argument can accept wildcards, or a file with the suffix .bam_list that contains the locations of multiple BAM files", minElements = 1)
+    public List<File> INPUT_BAM;
+
+    @Argument(shortName = StandardOptionDefinitions.SORT_ORDER_SHORT_NAME)
+    public SAMFileHeader.SortOrder SORT_ORDER = SAMFileHeader.SortOrder.coordinate;
+
+    @Argument(doc = "Flag indicating whether to check each input BAM file(s) for the presence of a paired-end read", shortName = "PR", optional = true)
+    public boolean CHECK_CONTAINS_PAIRED_READS = false;
+
+    @Override
+    protected int doWork() {
+        this.INPUT_BAM = FileListParsingUtils.expandFileList(INPUT_BAM);
+
+        File templateBamFile = null;
+        SAMSequenceDictionary templateDictionary = null;
+
+        StringBuilder errorMsg = new StringBuilder();
+        SamReaderFactory samReaderFactory = SamReaderFactory.makeDefault();
+        for (File inputBam : INPUT_BAM) {
+            try {
+                IOUtil.assertFileIsReadable(inputBam);
```

Comment:
If this assertion failed, don't you want to skip the subsequent tests for this file?